scripts/data_prep/convert_text_to_mds.py

```diff
@@ -0,0 +1,329 @@
+    # Compare the arguments
+    prev_args_str = done_file_contents[0]
+    if prev_args_str != args_str:
+        return False
+
+    # Compare file names
+    prev_names = done_file_contents[1:]
+    if len(prev_names) != len(object_names):
+        return False
+    for idx, prev_name in enumerate(prev_names):
+        if object_names[idx] != prev_name:
+            return False
+    return True
+
+
+def write_done_file(folder: str, file_name: str, args_str: str,
+                    object_names: List[str]):
+    with open(os.path.join(folder, file_name), 'w') as done_file:
+        done_file.write('\n'.join([args_str] + object_names) + '\n')
+
+
+def main(
+    tokenizer_name: str,
+    output_folder: str,
+    input_folder: str,
+    concat_tokens: int,
+    eos_text: str,
+    bos_text: str,
+    no_wrap: bool,
+    max_workers: int,
+    compression: str,
+    processes: int,
+    args_str: str,
+    reprocess: bool,
+):
+    done_file_name = 'done'
+    is_remote_output = is_remote_path(output_folder)
+
+    object_names = get_object_names(input_folder)
+
+    # Check if the text files in the bucket have already been processed.
+    if not reprocess and is_already_processed(output_folder, done_file_name,
+                                              args_str, object_names):
+        print(
+            f'Input folder {input_folder} is already processed at {output_folder}.'
+        )
+        return
+
+    # Use a temporary local directory if the output is remote and there are more than 1 processes
+    local_output_folder = tempfile.TemporaryDirectory(
+    ).name if is_remote_output else output_folder
+
+    if processes > 1:
+        # Download and convert the text files in parallel
+        args = get_task_args(object_names, local_output_folder, input_folder,
+                             processes, tokenizer_name, concat_tokens, eos_text,
+                             bos_text, no_wrap, compression, max_workers)
+        with Pool(processes=processes) as pool:
+            pool.starmap(download_and_convert, args)
+
+        # Merge the mds shards from each of the processes into a single folder
+        merge_shard_groups(local_output_folder)
+    else:
+        download_and_convert(object_names, local_output_folder, input_folder,
+                             tokenizer_name, concat_tokens, eos_text, bos_text,
+                             no_wrap, compression, max_workers)
+
+    # Write a done file with the args and object names
+    write_done_file(local_output_folder, done_file_name, args_str, object_names)
+
+    if is_remote_output:
+        # Upload the local output to the remote location
+        output_object_store = cast(
+            ObjectStore, maybe_create_object_store_from_uri(output_folder))
+        _, _, output_folder_prefix = parse_uri(output_folder)
+        pattern = os.path.join(local_output_folder, '*')
+        files_to_upload = sorted(glob(pattern))
+
+        # TODO: Use multi threading to upload files?
```

Comment:
we could do this, but the functions require state and logger args (init and upload_file). The usage of those args is pretty minimal and close to none...should we just pass dummy objects? I don't really like the idea of relying on understanding the internal implementation to use the functions because it could break easily with a change

Reply:
i think a better but more involved solution would be to refactor the composer code to move the logic into a class that doesn't implement LoggerDestination and the LoggerDestination version can call that class... wdyt?

We could keep this for now and then refactor in a separate PR(s)